examples/cluster/src/lib.rs

```diff
@@ -0,0 +1,294 @@
+use std::{
+    cell::{RefCell, RefMut},
+    collections::VecDeque,
+    error::Error,
+    future::Future,
+    rc::Rc,
+    time::Duration,
+};
+
+use rand::{rngs::SmallRng, seq::SliceRandom, Rng, SeedableRng};
+use tokio::{
+    io::{AsyncReadExt, AsyncWriteExt},
+    task::JoinSet,
+};
+use turmoil::{
+    net::{TcpListener, TcpStream},
+    Sim,
+};
+
+#[derive(Clone)]
+pub struct Cluster {
+    shared: Rc<RefCell<State>>,
+    rng: Rc<RefCell<SmallRng>>,
+}
+
+pub struct State {
+    machines: Vec<Machine>,
+    events: VecDeque<Event>,
+}
+
+pub enum Event {
+    Crash(String),
+    Spawn,
+}
+
+#[derive(Clone, Debug)]
+pub struct Machine {
+    pub name: String,
+    pub failed: bool,
+}
+
+impl Cluster {
+    pub fn new(servers: usize, seed: Option<u64>) -> Self {
+        let machines = (0..servers)
+            .map(|i| format!("server_{}", i))
+            .map(|name| Machine {
+                name,
+                failed: false,
+            })
+            .collect();
+
+        let seed = seed.unwrap_or_else(|| getrandom::u64().unwrap());
+
+        tracing::info!("starting turmoil sim with seed: {}", seed);
+
+        let rng = Rc::new(RefCell::new(SmallRng::seed_from_u64(seed)));
+
+        let state = State {
+            machines,
+            events: VecDeque::new(),
+        };
+
+        Cluster {
+            shared: Rc::new(RefCell::new(state)),
+            rng,
+        }
+    }
+
+    pub fn setup(&self, sim: &mut Sim<'_>) {
+        let me = self.shared.borrow();
+
+        for machine in me.machines.iter() {
+            let machine = machine.clone();
+            let name = machine.name.clone();
+            sim.host(name.as_str(), move || async move {
+                if let Err(e) = server("0.0.0.0:8080".into()).await {
+                    tracing::error!("server error: {:?}", e);
```

Comment:
nit (and throughout):
```suggestion
                    tracing::error!(?e, "server error");
```